boefjes/plugins/kat_manual/csv/normalize.py

```diff
@@ -0,0 +1,143 @@
+import csv
+import io
+import ipaddress
+import logging
+from typing import Union, Iterator, Dict, Tuple, List
+from octopoes.models import OOI, Reference
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.network import Network, IPAddressV4, IPAddressV6
+from octopoes.models.ooi.web import URL
+from octopoes.models.types import OOIType
+from pydantic import ValidationError
+
+from boefjes.job_models import NormalizerMeta
+
+
+OOI_TYPES = {
+    "Hostname": {"type": Hostname},
+    "URL": {"type": URL},
+    "Network": {"type": Network, "default": "internet", "argument": "name"},
+    "IPAddressV4": {"type": IPAddressV4},
+    "IPAddressV6": {"type": IPAddressV6},
+}
+
+logger = logging.getLogger(__name__)
+
+
+def run(normalizer_meta: NormalizerMeta, raw: Union[bytes, str]) -> Iterator[OOI]:
+    reference_cache = {"Network": {"internet": Network(name="internet")}}
+
+    yield from process_csv(raw, reference_cache)
+
+
+def process_csv(csv_raw_data, reference_cache):
+    csv_data = io.StringIO(csv_raw_data.decode("UTF-8"))
+
+    object_type = get_object_type(csv_data)
+
+    for row_number, row in enumerate(csv.DictReader(csv_data, delimiter=",", quotechar='"')):
+        if not row:
+            continue  # skip empty lines
+
+        try:
+            ooi, extra_declarations = get_ooi_from_csv(object_type, row, reference_cache)
+
+            for declaration in extra_declarations:
+                yield declaration
+
+            yield {
+                "type": "declaration",
+                "ooi": ooi.dict(),
+            }
+        except ValidationError:
+            logger.exception("Validation failed for row %s", row)
+
+
+def get_object_type(csv_data: io.StringIO) -> str:
+    csv_data.seek(0)
+    headers = csv_data.readline()
+    line = csv_data.readline()
+    csv_data.seek(0)
+
+    if "name" in headers:
+        return "Hostname"
+
+    if "raw" in headers:
+        return "URL"
+
+    if "address" in headers:
+        if line:
+            address = next(csv.DictReader(csv_data, delimiter=",", quotechar='"'))["address"]
+            csv_data.seek(0)
+
+            try:
```

Comment:
Use the function `ip_network` instead: this will take an IP string/int and return an object of the correct type (or raises a ValueError)